WireDomain/Tests/WireDomainTests/Event Processing/TeamEventProcessor/TeamMemberLeaveEventProcessorTests.swift

```diff
@@ -0,0 +1,331 @@
+
+            /// users won't be deleted as we might be in other (non-team) conversations with them
+            let team = try XCTUnwrap(Team.fetch(with: Scaffolding.teamID, in: context), "No team")
+
+            XCTAssertEqual(team.members, [])
+        }
+    }
+
+    func testProcessEvent_It_Request_Account_Deletion_When_Receiving_A_Team_Member_Leave_Update_Event_For_Self_User() async throws {
+        // Given
+
+        await context.perform { [context] in
+            let user = ZMUser.selfUser(in: context)
+            user.remoteIdentifier = Scaffolding.userID
+
+            let team = Team.insertNewObject(in: context)
+            team.remoteIdentifier = Scaffolding.teamID
+
+            let member = Member.getOrUpdateMember(
+                for: user,
+                in: team,
+                context: context
+            )
+
+            XCTAssertNotNil(member)
+            XCTAssertEqual(user.membership, member)
+        }
+
+        let expectation = XCTestExpectation()
+        let notificationName = AccountDeletedNotification.notificationName
+
+        NotificationCenter.default.addObserver(
+            forName: notificationName,
+            object: nil,
+            queue: nil
+        ) { notification in
+
+            XCTAssertNotNil(notification.userInfo?[notificationName] as? AccountDeletedNotification)
+
+            expectation.fulfill()
+        }
+
+        // When
+
+        try await sut.processEvent(Scaffolding.event)
+
+        // Then
+
+        await fulfillment(of: [expectation], timeout: 1)
+    }
+
+    func testProcessEvent_It_Removes_A_Member_From_All_Group_Conversations_They_Were_Part_Of_When_Receiving_A_Member_Leave_Event_For_That_Member() async throws {
+        // Given
+
+        await context.perform { [context] in
+            let user = ZMUser.insertNewObject(in: context)
+            user.remoteIdentifier = Scaffolding.userID
+
+            let otherUser = ZMUser.insertNewObject(in: context)
+            otherUser.remoteIdentifier = UUID()
+
+            let team = Team.insertNewObject(in: context)
+            team.remoteIdentifier = Scaffolding.teamID
+
+            let teamConversation1 = ZMConversation.insertNewObject(in: context)
+            teamConversation1.remoteIdentifier = Scaffolding.teamConversationID
+            teamConversation1.conversationType = .group
+            teamConversation1.addParticipantAndUpdateConversationState(
+                user: user,
+                role: nil
+            )
+            teamConversation1.team = team
+
+            let conversation = ZMConversation.insertGroupConversation(
+                moc: context,
+                participants: [user, otherUser]
+            )
+
+            conversation?.remoteIdentifier = Scaffolding.conversationID
+
+            let member = Member.getOrUpdateMember(
+                for: user,
+                in: team,
+                context: context
+            )
```

Comment:
The `ModelHelper` can help makes these models :)

Reply:
done